rpcclient/infrastructure.go

```diff
@@ -1656,7 +1665,15 @@ func (c *Client) BackendVersion() (BackendVersion, error) {
 	return c.backendVersion, nil
 }
 
-func (c *Client) sendAsync() FutureGetBulkResult {
+func (c *Client) sendAsync() (FutureGetBulkResult, error) {
+	c.batchLock.Lock()
+	defer c.batchLock.Unlock()
+
+	// if batchlist is empty, there's nothing to send
```

Comment:
nit (pre-existing from copy/paste below): Please use full sentences for comments (with initial capitalization and punctuation). Also, it should either say "batch list" or "batchList".